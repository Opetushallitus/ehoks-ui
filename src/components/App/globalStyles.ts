import { injectGlobal } from "emotion"

injectGlobal`
  html {
    box-sizing: border-box;
  }

  *, *:before, *:after {
    box-sizing: inherit;
  }

  html, body {
    margin: 0;
    padding: 0;
    background: #fff;
  }

  body  {
    font-family: 'Source Sans Pro', sans-serif;
  }

  /*
  * react-circular-progressbar styles
  *
  * All of the styles in this file are optional and configurable!
  */
  .CircularProgressbar .CircularProgressbar-path {
    stroke: #027FA9;
    stroke-linecap: square;
  }

  .CircularProgressbar .CircularProgressbar-trail {
    stroke: #d6d6d6;
  }

  .CircularProgressbar .CircularProgressbar-text {
    fill: #027FA9;
    font-size: 48px;
    dominant-baseline: middle;
    text-anchor: middle;
  }
`
