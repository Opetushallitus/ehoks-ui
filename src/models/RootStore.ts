import camelCase from "lodash.camelcase"
// import kebabCase from "lodash.kebabcase"
import mapObj from "map-obj"
import { flow, getEnv, types } from "mobx-state-tree"
import { EducationProviderStore } from "models/EducationProviderStore"
import { StudentStore } from "models/StudentStore"
import { WorkplaceProviderStore } from "models/WorkPlaceProviderStore"
import { LearningPeriod } from "./LearningPeriod"

const RootStoreModel = {
  activeLocale: types.optional(
    types.union(types.literal("fi"), types.literal("sv")),
    "fi"
  ),
  education: types.optional(EducationProviderStore, { info: {} }),
  isLoading: false,
  learningPeriods: types.optional(types.array(LearningPeriod), []),
  student: types.optional(StudentStore, { info: {} }),
  work: types.optional(WorkplaceProviderStore, { info: {} })
}

export const RootStore = types
  .model("RootStore", RootStoreModel)
  .views(self => ({
    // views defined here should be used from other stores using: getRoot(self)
    get fetch() {
      return getEnv(self).fetch
    }
  }))
  .actions(self => {
    const fetchSingle = flow(function*(url: string) {
      const response = yield self.fetch(url)
      const model = response.data && response.data[0] ? response.data[0] : {}
      // camelCase kebab-cased object keys recursively so we can use dot syntax for accessing values
      return mapObj(
        model,
        (key: string, value: any) => [camelCase(key), value],
        {
          deep: true
        }
      )
    })

    const fetchCollection = flow(function*(url: string) {
      const response = yield self.fetch(url)
      console.log("fetchCollection", response)
      const model = response.data ? response.data : []
      return model
    })

    return { fetchSingle, fetchCollection }
  })
