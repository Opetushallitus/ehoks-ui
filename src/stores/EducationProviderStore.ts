import { apiUrl } from "config"
import { flow, getRoot, types } from "mobx-state-tree"
import {
  EducationProviderInfo,
  IEducationProviderInfo
} from "models/EducationProviderInfo"
import { ApiResponse, IRootStore } from "stores/RootStore"

const EducationProviderModel = {
  info: types.optional(EducationProviderInfo, {}),
  isLoading: false
}

export const EducationProviderStore = types
  .model("EducationProviderStore", EducationProviderModel)
  .actions(self => {
    const root = getRoot<IRootStore>(self)

    const fetchInfo = flow(function*(): any {
      self.isLoading = true
      const response: ApiResponse<
        IEducationProviderInfo
      > = yield root.fetchSingle(apiUrl("education/info/"))
      self.info = response.data
      self.isLoading = false
    })

    return { fetchInfo }
  })
