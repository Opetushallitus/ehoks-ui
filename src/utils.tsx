import React from "react"
import { IntlProvider } from "react-intl"
import renderer from "react-test-renderer"
import { translations } from "./translations"

// mocked fetch using local json files
export const mockFetch = (url: string) => {
  const [, path] = url.split("http://localhost:3000/")
  return Promise.resolve(
    require("./models/mocks/" + path.replace(/\//g, "_") + ".json")
  )
}

// fetch that returns the JSON directly
export const fetch = (url: string) =>
  window.fetch(url).then(response => response.json())

function defaultCreateNodeMock(): null {
  return null
}

export const createComponentWithIntl = (
  children: React.ReactNode,
  createNodeMock?: () => any,
  props = { locale: "fi" }
) => {
  const options = { createNodeMock: createNodeMock || defaultCreateNodeMock }
  return renderer.create(
    <IntlProvider {...props} messages={translations[props.locale]}>
      {children}
    </IntlProvider>,
    options
  )
}
