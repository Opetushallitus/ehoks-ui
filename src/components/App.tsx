import { Router } from "@reach/router"
import { GlobalStyles } from "components/App/globalStyles"
import { AppFooter } from "components/AppFooter"
import { AppHeader } from "components/AppHeader"
import { inject, observer } from "mobx-react"
import React from "react"
import { IntlProvider } from "react-intl"
import { Ammatillinentutkinto } from "routes/Ammatillinentutkinto"
import { Henkilokohtaistaminen } from "routes/Henkilokohtaistaminen"
import { TyopaikanToimija } from "routes/TyopaikanToimija"
import { IRootStore } from "stores/RootStore"
import styled from "styled"
import { Etusivu } from "../routes/Etusivu"
import { AppErrors } from "./AppErrors"
import { ThemeWrapper } from "./ThemeWrapper"

const Container = styled("div")`
  margin: 0;
`

export interface AppProps {
  store?: IRootStore
}

@inject("store")
@observer
export class App extends React.Component<AppProps> {
  componentDidMount() {
    // load user session info from backend
    this.props.store!.session.checkSession()
  }

  render() {
    const { store } = this.props
    return (
      <ThemeWrapper>
        <IntlProvider
          defaultLocale="fi"
          locale={store!.translations.activeLocale}
          messages={
            store!.translations.messages[store!.translations.activeLocale]
          }
          textComponent={React.Fragment}
        >
          <Container>
            <AppHeader />
            <AppErrors />
            <Router basepath="/ehoks">
              <Etusivu path="/*" />
              <TyopaikanToimija path="tyopaikantoimija" />
              <Henkilokohtaistaminen path="henkilokohtaistaminen" />
              <Ammatillinentutkinto path="ammatillinentutkinto" />
            </Router>
            <AppFooter />
            <GlobalStyles />
          </Container>
        </IntlProvider>
      </ThemeWrapper>
    )
  }
}
