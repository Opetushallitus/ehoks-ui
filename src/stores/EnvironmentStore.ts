import { apiUrl } from "config"
import { flow, getRoot, Instance, types } from "mobx-state-tree"
import { IRootStore } from "stores/RootStore"

const EnvironmentStoreModel = {
  eperusteetPerusteUrl: types.optional(types.string, ""),
  error: types.optional(types.string, ""),
  isLoading: false,
  opintopolkuLoginUrl: types.optional(types.string, ""),
  opintopolkuLogoutUrl: types.optional(types.string, "")
}

export const EnvironmentStore = types
  .model("EnvironmentStore", EnvironmentStoreModel)
  .actions(self => {
    const root = getRoot<IRootStore>(self)

    const getEnvironment = flow(function*(): any {
      self.isLoading = true
      try {
        const response = yield root.fetchSingle(apiUrl("misc/environment"))
        const {
          eperusteetPerusteUrl,
          opintopolkuLoginUrl,
          opintopolkuLogoutUrl
        } = response.data
        self.eperusteetPerusteUrl = eperusteetPerusteUrl
        self.opintopolkuLoginUrl = opintopolkuLoginUrl
        self.opintopolkuLogoutUrl = opintopolkuLogoutUrl
      } catch (error) {
        root.errors.logError(
          "EnvironmentStore.getEnvironment",
          "Ympäristömuuttujien haku epäonnistui",
          error.message
        )
      }
      self.isLoading = false
    })

    return { getEnvironment }
  })

export interface IEnvironmentStore extends Instance<typeof EnvironmentStore> {}
