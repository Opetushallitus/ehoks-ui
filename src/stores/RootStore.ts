// import kebabCase from "lodash.kebabcase"
import { getEnv, Instance, types } from "mobx-state-tree"
import { EducationProviderStore } from "stores/EducationProviderStore"
import { EnvironmentStore } from "stores/EnvironmentStore"
import { OppilasStore } from "stores/OppilasStore"
import { SessionStore } from "stores/SessionStore"
import { TranslationStore } from "stores/TranslationStore"
import { TyopaikanToimijaStore } from "stores/TyopaikanToimijaStore"

export interface ApiResponse<T> {
  meta: {
    [name: string]: any
  }
  data: T
}

export interface InjectedStores {
  store: IRootStore
}

const RootStoreModel = {
  education: types.optional(EducationProviderStore, { info: {} }),
  environment: types.optional(EnvironmentStore, {}),
  isLoading: false,
  oppilas: types.optional(OppilasStore, {}),
  session: types.optional(SessionStore, {}),
  translations: types.optional(TranslationStore, {}),
  tyopaikanToimija: types.optional(TyopaikanToimijaStore, {})
}

export const RootStore = types
  .model("RootStore", RootStoreModel)
  .views(self => {
    return {
      get errors() {
        return getEnv(self).errors
      }
    }
  })

export interface IRootStore extends Instance<typeof RootStore> {}
