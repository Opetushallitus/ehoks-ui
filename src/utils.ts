import { apiUrl } from "config"
import { Instance } from "mobx-state-tree"
import { EnvironmentStore } from "stores/EnvironmentStore"
import { InjectedStores, RootStore } from "stores/RootStore"
import { TranslationStore } from "stores/TranslationStore"
import "whatwg-fetch" // polyfill window.fetch for IE 11

// mocked fetch using local json files
export const mockFetch = (version = 0) => (url: string) => {
  const [, path] = url.split(apiUrl(""))
  return Promise.resolve({
    json: () => {
      const json = require("./stores/mocks/" +
        path.replace(/\/|\-/g, "_") +
        version +
        ".json")
      return Promise.resolve(json)
    },
    ok: true,
    statusText: "Error"
  })
}

// fetch that includes credentials
export const fetch = (url: string | Request, init: RequestInit) =>
  window.fetch(url, { credentials: "include", ...init })

export const injectSession = (stores: InjectedStores) => ({
  session: stores.store.session
})

export const getTranslations = (
  root: Instance<typeof RootStore>
): Instance<typeof TranslationStore> => {
  return root.translations
}

export const getEnvironment = (
  root: Instance<typeof RootStore>
): Instance<typeof EnvironmentStore> => {
  return root.environment
}
