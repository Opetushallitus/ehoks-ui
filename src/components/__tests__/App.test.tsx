import { Provider } from "mobx-react"
import React from "react"
import renderer from "react-test-renderer"
import { RootStore } from "stores/RootStore"
import { mockFetch } from "../../utils"
import { App } from "../App"

it("App renders correctly", async () => {
  // pass fetch to root store using DI, so we can easily mock it in tests
  const store = RootStore.create({}, { fetch: mockFetch() })
  // fetch default translations
  await store.translations.haeLokalisoinnit()
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
