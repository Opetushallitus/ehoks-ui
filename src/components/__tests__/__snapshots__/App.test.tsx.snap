// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="css-r44k6v"
>
  <header
    className="css-1c9pu1m"
  >
    <div
      className="css-1bg6mpz"
    >
      <nav
        className="css-1lhzg4m"
      >
        <a
          className="css-fxt1xk"
          href="https://opintopolku.fi/"
        >
          Opintopolku.fi
        </a>
        <a
          className="css-fxt1xk"
          href="https://opintopolku.fi/oma-opintopolku/"
        >
          Oma Opintopolku
        </a>
        <a
          className="css-fxt1xk"
          href="https://eperusteet.opintopolku.fi/"
        >
          ePerusteet
        </a>
        <a
          className="css-qn17of"
          href="/"
        >
          eHOKS
        </a>
      </nav>
    </div>
    <div
      className="css-1xu5u1k"
    >
      <div
        className="css-1oo5bs0"
      >
        <svg
          fill="currentColor"
          height="40"
          size="40"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="40"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <h3>
          Valikko
        </h3>
      </div>
      <div
        className="css-fmwec3"
      >
        eHOKS
      </div>
    </div>
  </header>
  <div
    className="css-abysv6"
  />
  <footer
    className="css-cqueuz"
  >
    <div
      className="css-1twx585"
    >
      <div
        className="css-1dnql38"
      >
        <div
          className="css-1qxls6i"
        >
          <img
            alt="Opintopolku logo"
            className="css-29yp7r"
            src=""
          />
          <div
            className="css-139jliq"
          >
            <a>
              <img
                alt="Twitter"
                className="css-1rcxayq"
                src=""
              />
            </a>
            <a>
              <img
                alt="Facebook"
                className="css-1rcxayq"
                src=""
              />
            </a>
            <a>
              <img
                alt="Instagram"
                className="css-1rcxayq"
                src=""
              />
            </a>
          </div>
        </div>
        <ul
          className="css-1fmvbse"
        >
          <li>
            ePerusteet
          </li>
          <li>
            Mikä on opintopolku
          </li>
          <li>
            Tietoturvaseloste
          </li>
          <li>
            Anna palautetta
          </li>
        </ul>
      </div>
      <div
        className="css-os38x1"
      >
        <div
          className="css-of0swa"
        >
          <img
            alt="Opetus- ja kulttuuriministeriö logo"
            className="css-14qd04p"
            src=""
          />
          <img
            alt="Opetushallitus logo"
            className="css-14qd04p"
            src=""
          />
        </div>
        <div
          className="css-nadzu"
        >
          <p>
            Koulutuksen järjestäjät ja korkeakoulut ylläpitävät tietoja
              koulutuksistaan Opintopolku.fi-palvelussa. Tietojen oikeellisuuden
              voit tarkistaa kyseisestä oppilaitoksesta tai korkeakoulusta.
          </p>
          <div
            className="css-1hgh9qx"
          >
            <div
              className="css-fth9yi"
            >
              <a>
                Suomeksi
              </a>
              <a>
                In English
              </a>
              <a>
                På svenska
              </a>
            </div>
            <div
              className="css-1cmi7g3"
            >
              |
            </div>
            <div
              className="css-hx8h8z"
            >
              <a>
                <img
                  alt="Twitter"
                  className="css-1rcxayq"
                  src=""
                />
              </a>
              <a>
                <img
                  alt="Facebook"
                  className="css-1rcxayq"
                  src=""
                />
              </a>
              <a>
                <img
                  alt="Instagram"
                  className="css-1rcxayq"
                  src=""
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
